solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -157,20 +168,16 @@ public void call(SolrQueryRequest req, SolrQueryResponse rsp) {
         origCorename = pathSegments.get(1);
         core = cores.getCore(origCorename);
       }
+
+      // We didn't find a core, so we're either an error or a Jersey 'ADMIN' api
       if (core == null) {
-        log.error(">> path: '{}'", path);
-        if (path.endsWith(CommonParams.INTROSPECT)) {
-          initAdminRequest(path);
-          return;
-        } else {
-          throw new SolrException(
-              SolrException.ErrorCode.NOT_FOUND,
-              "no core retrieved for core name: " + origCorename + ". Path: " + path);
-        }
-      } else {
-        Thread.currentThread().setContextClassLoader(core.getResourceLoader().getClassLoader());
+        // This codepath needs to cover custom plugin Jersey APIs, as well as in-built ones
+        log.info("Hoping this path is covered by the Jersey-admin case: {}", path);
```

Comment:
I recommend use of "nocommit" for this and anything else that's not supposed to be merged